packages/ember-data/lib/serializers/json-serializer.js

```diff
@@ -111,6 +111,19 @@ export default Serializer.extend({
   mergedProperties: ['attrs'],
 
   /**
+    The `_mustSerialize` property is set to true when the user
+    explicitely use `{serialize: true}` when defining the attrs hash.
+
+    Specifically it takes priority over the other checks and the related
+    attribute/relationship will be serialized
+
+    @property _mustSerialize
+    @private
+  */
+  _mustSerialize: false,
```

Comment:
This doesn't make sense as a flag, it's a value that changes for each key, so needs to be a function, but we can just make it a part of canSerialize